pkg/cephconfig/config.go

```diff
@@ -0,0 +1,437 @@
+	}
+	return paramInfo, nil
+}
+
+// Search searches for configuration parameters according to the query parameters
+func (c *Config) Search(query QueryParams) []ConfigParamInfo {
+	// Set default values
+	if query.Sort == 0 {
+		query.Sort = pb.SearchConfigRequest_NAME
+	}
+	if query.Order == 0 {
+		query.Order = pb.SearchConfigRequest_ASC
+	}
+
+	// Remove pre-allocation optimization
+	var result []ConfigParamInfo
+
+	// Convert full text to lowercase once if needed
+	var fullTextLower string
+	if query.FullText != "" {
+		fullTextLower = strings.ToLower(query.FullText)
+	}
+
+	// Filter parameters
+	for _, info := range c.params {
+		if !c.matchesService(info, query.Service) {
+			continue
+		}
+		if !c.matchesName(info, query.Name) {
+			continue
+		}
+		if !c.matchesLevel(info, query.Level) {
+			continue
+		}
+		if !c.matchesFullText(info, fullTextLower) {
+			continue
+		}
+		result = append(result, info)
+	}
+
+	// Sort results using Go's built-in sort
+	c.sortResults(result, query.Sort, query.Order)
+
+	return result
+}
+
+// matchesService checks if the parameter matches the service filter
+func (c *Config) matchesService(info ConfigParamInfo, service pb.SearchConfigRequest_ServiceType) bool {
+	if service == pb.SearchConfigRequest_COMMON {
+		return true
+	}
+
+	// Map enum to canonical string representation
+	var serviceStr string
+	switch service {
+	case pb.SearchConfigRequest_COMMON:
+		serviceStr = "common"
+	case pb.SearchConfigRequest_MON:
+		serviceStr = "mon"
+	case pb.SearchConfigRequest_MDS:
+		serviceStr = "mds"
+	case pb.SearchConfigRequest_OSD:
+		serviceStr = "osd"
+	case pb.SearchConfigRequest_MGR:
+		serviceStr = "mgr"
+	case pb.SearchConfigRequest_RGW:
+		serviceStr = "rgw"
+	case pb.SearchConfigRequest_RBD:
+		serviceStr = "rbd"
+	case pb.SearchConfigRequest_RBD_MIRROR:
+		serviceStr = "rbd-mirror"
+	case pb.SearchConfigRequest_IMMUTABLE_OBJECT_CACHE:
+		serviceStr = "immutable-object-cache"
+	case pb.SearchConfigRequest_MDS_CLIENT:
+		serviceStr = "mds_client"
+	case pb.SearchConfigRequest_CEPHFS_MIRROR:
+		serviceStr = "cephfs-mirror"
+	case pb.SearchConfigRequest_CEPH_EXPORTER:
+		serviceStr = "ceph-exporter"
+	default:
+		serviceStr = service.String()
+	}
+
+	for _, svc := range info.Services {
+		if strings.EqualFold(svc, serviceStr) {
+			return true
+		}
+	}
+	return false
+}
+
+// matchesName checks if the parameter matches the name filter
+func (c *Config) matchesName(info ConfigParamInfo, name string) bool {
+	if name == "" {
+		return true
+	}
+	return matchWildcard(info.Name, name)
+}
+
+// matchesLevel checks if the parameter matches the level filter
+func (c *Config) matchesLevel(info ConfigParamInfo, level pb.SearchConfigRequest_ConfigLevel) bool {
+	if level == pb.SearchConfigRequest_BASIC {
+		return strings.EqualFold(info.Level, "basic")
+	}
+	if level == pb.SearchConfigRequest_ADVANCED {
+		return strings.EqualFold(info.Level, "advanced")
```

Comment:
what is the purpose of this line? should it be enough to just compare enum values?